src/components/core/NotifyObservers.react.js

```diff
@@ -56,9 +60,30 @@ function NotifyObserversComponent({
 
     dependencies,
 
+    loading,
+
     fireEvent,
     setProps,
+    requestQueue,
 }) {
+    // loading prop coming from TreeContainer
+    let isLoading = loading;
+    let loadingProp;
+    let loadingComponent;
+
+    requestQueue.map(r => {
+        if (r.status === 'loading' && contains(id, r.controllerId)) {
+            isLoading = true;
+            loadingComponent = r.controllerId.split('.')[0];
+            loadingProp = r.controllerId.split('.')[1];
```

Comment:
Can split only one time: `[loadingProp, loadingComponent] = r.controllerId.split('.')`. Also wouldn't we only have the last loadingProp and loadingComponent in the requestQueue if there's multiple callback updating ?

Reply:
Had a hard time figuring this out, that is not a proper map. It should be a for loop or a filter or a find.

Reply:
the most appropriate would be a `.forEach()`